python/lsst/ts/linearstage/mocks/mock_zaber_lst.py

```diff
@@ -19,14 +19,315 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
-__all__ = ["MockSerial", "MockLST"]
+__all__ = ["MockSerial", "MockLST", "LinearStageServer", "MockLSTV2"]
 
+import asyncio
 import inspect
 import logging
 import queue
+import types
 
 import serial
-from zaber.serial import AsciiCommand
+from lsst.ts import simactuators, tcpip
+
+
+class LinearStageServer(tcpip.OneClientReadLoopServer):
+    """Implment the mock linearstage server.
+
+    Parameters
+    ----------
+    port : `int | None`
+        The port of the mock server.
+    log : `logging.Logger`
+        The log of the mock server.
+
+    Attributes
+    ----------
+    device : `MockLST`
+        The mock device that handles replies.
+    """
+
+    def __init__(self, port: int | None, log: logging.Logger) -> None:
+        super().__init__(
+            port=port,
+            host=tcpip.LOCAL_HOST,
+            log=log,
+            name="Zaber Mock Server",
+            terminator=b"\n",
+        )
+        self.device: MockLSTV2 = MockLSTV2()
+
+    async def read_and_dispatch(self) -> None:
+        """Read from the client and send a reply."""
+        try:
+            command: str = await self.read_str()
+        except asyncio.IncompleteReadError as e:
+            # Used for debugging received message
+            self.log.info(f"{e.partial}")
+            raise
+        self.log.info(f"{command=} received.")
+        reply: str = self.device.parse_message(command)
+        self.log.debug(f"Writing {reply=}.")
+        await self.write_str(reply)
+
+
+class MockLSTV2:
+    """Mock the zaber stage.
+
+    Attributes
+    ----------
+    address : `int`
+        The device address.
+    id : `int`
+        The ID number of the device.
+    system_serial : `int`
+        The serial number of the system.
+    version : `str`
+        The version string of the firmware.
+    version_build : `int`
+        The version build number of the firmware.
+    modified : `bool`
+        Has the device been modified?
+    axis_count : `int`
+        The number of axes attached to the device.
+    message_id : `int` | `None`
+        The ID of the current message.
+    max_packet : `int`
+        The number of bytes allowed in a packet.
+    max_word : `int`
+        The max size of a word.
+    position : `lsst.ts.simactuators.PointToPointActuator`
+        The encoder position along the stage.
+    axes : `types.SimpleNamespace`
+        The information from each axis.
+    """
+
+    def __init__(self) -> None:
+        self.address: int = 1
+        self.id: int = 30342
```

Comment:
Lots of magic numbers again. Please extract to constants. Also, some of them have very specific values. Is that necessary to make the library work well? If yes, please explain that in inline comments.